app/models/piece.rb

```diff
@@ -35,8 +35,6 @@ def check_mate?
   def valid_move?(new_row, new_column)
     return false if out_of_boundary?(new_row, new_column)
     return false if no_move?(new_row, new_column)
-    #return false unless proper_length?(new_row, new_column)
-    true
```

Comment:
Good question, I was thinking the same thing. I don't think any piece will just call valid_move? from the piece file. Thus, I agree that we can probably get rid of true here because the super valid_move? will only tell us if a move is wrong, not if it is completely correct.